src/View/ViewableData.php

```diff
@@ -346,33 +298,38 @@ public function customise($data)
      *
      * This method should be overridden in subclasses to provide more context about the classes state. For example, a
      * {@link DataObject} class could return false when it is deleted from the database
-     *
-     * @return bool
      */
-    public function exists()
+    public function exists(): bool
     {
         return true;
     }
 
     /**
-     * @return string the class name
+     * Output this model in a format that is ready to be rendered directly into the view layer.
      */
-    public function __toString()
+    public function __toString(): string
     {
-        return static::class;
+        return (string) $this->forTemplate();
     }
 
     /**
-     * @return ViewableData
+     * Output this model in a format that is ready to be rendered directly into the view layer
+     * (e.g. into a parent template).
+     *
+     * Most commonly this is done by rendering the model using an appropriate template.
+     *
+     * @throws NotImplementedException if not implemented by a subclass.
      */
-    public function getCustomisedObj()
+    public function forTemplate(): ?string
+    {
+        throw new NotImplementedException('This method should be implemented in subclasses');
+    }
```

Comment:
Intentionally explicitly adding this method so we have a known method signature for it - but then throwing an exception if it's not implemented to match existing behaviour.

Reply:
On second thought I'll do this as follow-up work